node/src/PipeTransport.ts

```diff
@@ -411,13 +414,18 @@ export class PipeTransport
 
 	private handleWorkerNotifications(): void
 	{
-		this.channel.on(this.internal.transportId, (event: string, data?: any) =>
+		this.channel.on(this.internal.transportId, (event: FbsEvent, data?: any) =>
 		{
 			switch (event)
 			{
-				case 'sctpstatechange':
+				case FbsEvent.TRANSPORT_SCTP_STATE_CHANGE:
 				{
-					const sctpState = data.sctpState as SctpState;
+					const notification = new FbsTransport.SctpStateChangeNotification();
+
+					(data as FbsNotification.Notification).body(notification);
+
+					const sctpState =
+						FbsSctpState.SctpState[notification.sctpState()].toLowerCase() as SctpState;
```

Comment:
Why is `toLowerCase()` needed here?

Reply:
`FbsSctpState` is an enum with the possible `sctpstate` 's:


```ts
export enum SctpState {
  NEW = 0,
  CONNECTING = 1,
  CONNECTED = 2,
  FAILED = 3,
  CLOSED = 4
}
`````

In order to get the `String` of the key of an enum value you can do it as follows:

```ts
const state = SctpState.NEW;
// Get the string value of the key pointed by the above enum value
const stateString = SctpState[state] // This points to a this string => "NEW".
````

I'm downcasing this so we have `new`, instead of `NEW` which is how we offer in the API.

Would it be more readable to have a helper function `SctpState2String(SctpState: state): string` that returns the lower value as desired?

Reply:
> Would it be more readable to have a helper function SctpState2String(SctpState: state): string that returns the lower value as desired?

Where would you put it?

Reply:
> Where would you put it?

`SctpState` is defined in `Transport.ts`, so I would put it there.

Reply:
We have similar utils in C++ code so I'd like having the same in JS.

Reply:
yep